syntax = "proto3";

message Item {
  string desc = 1;
  float value = 2;
}

message InsertRequest {
  int32 key = 1;
  string desc = 2;
  float value = 3;
}

message InsertResponse {
  int32 status = 1;
}

message QueryRequest {
  int32 key = 1;
}

message QueryResponse {
  Item item = 1;
}

message RegisterRequest {
  string server_name = 1;
  int32 server_port = 2;
}

message RegisterResponse {
  int32 result = 1;
}

message TerminateRequest {}

message TerminateResponse {
  int32 num_keys = 1;
}

service Directory {
  rpc Insert(InsertRequest) returns (InsertResponse) {}
  rpc Query(QueryRequest) returns (QueryResponse) {}
  rpc Register(RegisterRequest) returns (RegisterResponse) {}
  rpc Terminate(TerminateRequest) returns (TerminateResponse) {}
}
